### PR TITLE
 bpo-26510: Add versionchanged for required arg of add_subparsers

### PR DESCRIPTION
**Pull request by adamjstewart**
The `required` argument to `argparse.add_subparsers` was added in #3027. This PR specifies the earliest version of Python where it is available.

https://bugs.python.org/issue26510

Automerge-Triggered-By: @merwok

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@adamjstewart

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/library/argparse.rst
Comment (merwok) [inline Doc/library/argparse.rst]:
```suggestion
   .. versionchanged:: 3.7
```

If I recall correctly, the version* directive apply to their parent object block (module, class, function, data), so this is more correct.  We don’t want `add_subparsers` to show up in a list of functions added in 3.7 :slightly_smiling_face:

**Review by merwok — CHANGES_REQUESTED** [review]
Thanks for adding this piece that was missed in the original PR!

Because this part of the docs is super long, could you also add something like `(added in 3.7)` on line 1639?  Would help readers get the version info right when they read about `required`.

File: Doc/library/argparse.rst
Comment (adamjstewart) [reply Doc/library/argparse.rst]: I just copied this syntax from https://docs.python.org/3/library/argparse.html#argparse.FileType, should I fix that too while I'm at it?

Comment (adamjstewart) [reply Doc/library/argparse.rst]: https://www.sphinx-doc.org/en/1.5/markup/para.html#directive-versionadded seems inconclusive. They use `versionadded` and `versionchanged` for "new parameter" examples. I'll defer to your judgement on this one.

**Comment by merwok** [comment]
The what’s new note was removed by #7609 after the default value was changed in #6919, but now I realize that this was too much: subparsers are not required, but there is a new param!  Would you @adamjstewart mind re-adding a note in what’s new for 3.7?

(also please say if you need any help with the CLA)

**Comment by adamjstewart** [comment]
CLA has been signed 😄

**Comment by merwok** [comment]
Docs are a little vague indeed!  From https://devguide.python.org/documenting/#paragraph-level-markup

> **versionadded**
>    This directive documents the version of Python which added the described feature, or a part of it, to the library

But:

>  **versionchanged**
>     Similar to versionadded, but describes when and what changed in the named feature in some way (new parameters, changed side effects, platform support, etc.)

`new parameters` seems pretty clear here :slightly_smiling_face: 

I wouldn’t change FileType or other files in this PR.

File: Doc/library/argparse.rst
Comment (merwok) [inline Doc/library/argparse.rst]:
```suggestion
     ``False`` (added in 3.7)
```

Other parameters don’t have final period (and if they had, would be after the parens I think)

File: Doc/whatsnew/3.7.rst
Comment (merwok) [inline Doc/whatsnew/3.7.rst]: I think this focuses on the mecanism without explaining the use case.
What do you think of:

```
* :mod:`argparse` subparsers can now be made mandatory by passing ``required=True``
  to :meth:`ArgumentParser.add_subparsers() <argparse.ArgumentParser.add_subparsers>`
```

**Comment by merwok** [comment]
A note about the review: these are all quite minor changes!  I could do them myself by adding commits to the branch directly, but in general I prefer to have a discussion and let the contributor own their PR, unless they are not interested in or don’t have for tiny details, in which case I’m happy to polish and merge :slightly_smiling_face:

**Review by merwok — APPROVED** [review]

File: Doc/whatsnew/3.7.rst
Comment (merwok) [inline Doc/whatsnew/3.7.rst:2405]: Small thing I didn’t notice before merge: in func/meth/etc roles, we don’t include the parentheses.  Sphinx has a setting to add them or not automatically to all roles (I think it’s smart enough to not add them if already present, but not 100% sure, will check when the docs rebuild is published).

(I personnally don’t like to have them, especially given Python’s clear difference of referencing a function vs. calling it, but many programmers are used to writing things like `the whatever() function` probably to make it less ambiguous in plain text contexts like email, so Python’s doc is set up to have them :slightly_smiling_face:)

I didn’t pay attention to the fact that the section here (`Changes in the Python API`) was part of `Porting to Python 3.7` (we really need docs previews for PRs!), which made sense in the original PR (changed argparse behaviour) but less so now.  I think it would have been better in `Improved Modules` → `argparse` https://docs.python.org/3.9/whatsnew/3.7.html#argparse

**Pull request merged**
